packages/engine-canvas/index.js

```diff
@@ -1,14 +1,31 @@
-import { Mechanic } from "@mechanic-design/core";
+import { Mechanic } from '@mechanic-design/core';
 
-const root = document.getElementById("root");
+const root = document.getElementById('root');
 
 export const run = (functionName, func, values, config) => {
   const { isPreview } = config;
 
-  root.innerHTML = "";
+  root.innerHTML = '';
 
   const mechanic = new Mechanic(func.settings, values, config);
 
+  // Preview is always going to use the pixel density of the
+  // screen we're currently on.
+  //
+  // Export will respect the density setting made by the user
+  const canvasDensity = isPreview
+    ? window.devicePixelRatio || 1
+    : mechanic.exportDensity;
+
+  const preparedCanvas = document.createElement('canvas');
+  const preparedCanvasContext = preparedCanvas.getContext('2d');
+
+  if (values.width) preparedCanvas.width = values.width;
+  if (values.height) preparedCanvas.height = values.height;
+
+  preparedCanvas.style.transform = `scale${1 / canvasDensity}`;
+  preparedCanvasContext.scale(canvasDensity, canvasDensity);
```

Comment:
Is this canvas being used at all?

Reply:
Oh, good catch. That's leftover. Removed it